pathd/path_pcep_controller.c

```diff
@@ -377,7 +405,7 @@ int pcep_thread_get_counters_callback(struct thread *t)
 	assert(ctrl_state != NULL);
 	struct pcc_state *pcc_state;
 
-	if (args->pcc_id <= ctrl_state->pcc_count) {
+	if (args->pcc_id <= MAX_PCC) {
```

Comment:
There shouldn't be any more callback using an integer index to do anything.

Reply:
I'll keep and array to hold data and the id now is also and int but it's a sequential one no related with array index